zad2/warriors.rb

```diff
@@ -1,12 +1,58 @@
-class Warrior
-  def initialize(name:, level:)
-  end
+class Character
+	attr_reader :name
+	attr_reader :level
+	def initialize(name:,level:)
+		@name=name
+		@level=[level,1].max
+	end
+	def strength
+		return @level+[*1..12].sample
+  	end
 
-  def strength
-  end
+	def card
+		return "#{@name} (lvl #{@level})"
+	end
+end
+
+class Warrior < Character
+	def level_up(level)
+		@level=[@level+level,99].min
+	end
+end
+
+class Monster < Character
 end
 
 class BattleArena
-  def initialize(first_character, second_character)
-  end
+  	def initialize(first_character, second_character)
+		@first_character=first_character
+		@second_character=second_character
+  	end
+
+	def alert(first,second,strength)
+		puts "#{first.name} attacked #{second.name} with #{strength} damage" 
+	end
+
+	def winner(character,level)
+		puts "Won #{character.name}"
+		if character.is_a? Warrior and level>0
```

Comment:
proponuję używać `&&` i `||` zamiast `and` i `or` ze względu to trochę inne priorytety wykonywania działań.